lib/money/bank/base.rb

```diff
@@ -93,14 +93,15 @@ def setup
       # @param [Money] from The +Money+ object to exchange from.
       # @param [Money::Currency, String, Symbol] to_currency The currency
       #  string or object to exchange to.
+      # @param [Numeric] rate Rate to use when exchanging currencies.
       # @yield [n] Optional block to use to round the result after making
       #  the exchange.
       # @yieldparam [Float] n The result after exchanging from one currency to
       #  the other.
       # @yieldreturn [Integer]
       #
       # @return [Money]
-      def exchange_with(from, to_currency, &block)
+      def exchange_with(from, to_currency, rate: nil, &block)
```

Comment:
I just realized this is going to break the implementation of a bank that exists today. Sorry, I think we are going to need to create a new method that can be automatically be not implemented for existing banks.

So, let's try and make a new method called `#exchange_with_using_custom_rate(from, to_currency, rate, &block)`

Reply:
My bad, I didn't think about that when I wrote it.